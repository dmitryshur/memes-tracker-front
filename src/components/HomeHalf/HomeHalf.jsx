import React from 'react';
import PropTypes from 'prop-types';
import './HomeHalf.scss';

const HomeHalf = ((props) => {
  return (
    <div className={`home-half home-half-${props.side}`}>
      <div className={`home-overlay home-overlay-${props.side}`} />
    </div>
  );
});

HomeHalf.propTypes = {
  side: PropTypes.string
};

HomeHalf.defaultProps = {
  side: 'left'
};

export default HomeHalf;
