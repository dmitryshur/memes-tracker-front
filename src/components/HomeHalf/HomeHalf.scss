$newest-bg: #47a3da;
$liked-bg: #ffffff;

.home-half {
  height: 100%;
  opacity: 0.95;
  width: 50%;
  z-index: 1;
  position: relative;

  .home-overlay {
    height: 100%;
    position: absolute;
    width: 100%;
    z-index: 0;

    &-newest {
      background-color: $newest-bg;
    }

    &-liked {
      background-color: $liked-bg;
    }
  }

}
