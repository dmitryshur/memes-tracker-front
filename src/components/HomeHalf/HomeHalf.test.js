import React from 'react';
import { shallow } from 'enzyme';
import { HomeHalf } from './HomeHalf.jsx';

test('testing HomeHalf newest component', () => {
  const wrapper = shallow(<HomeHalf section="newest" />);
  expect(wrapper.find('.home-half-newest').length).toBeGreaterThan(0);
  expect(wrapper.find('.home-overlay-newest').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf liked component', () => {
  const wrapper = shallow(<HomeHalf section="liked" />);
  expect(wrapper.find('.home-half-liked').length).toBeGreaterThan(0);
  expect(wrapper.find('.home-overlay-liked').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf newest component with default state', () => {
  const wrapper = shallow(<HomeHalf section="newest" />);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf newest component with opened state', () => {
  const wrapper = shallow(<HomeHalf section="newest" chosen={true} />);
  expect(wrapper.find('.opened').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf newest component with closed state', () => {
  const wrapper = shallow(<HomeHalf section="newest" chosen={false} />);
  expect(wrapper.find('.closed').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf liked component with default state', () => {
  const wrapper = shallow(<HomeHalf section="liked" />);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf liked component with opened state', () => {
  const wrapper = shallow(<HomeHalf section="liked" chosen={true} />);
  expect(wrapper.find('.opened').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('testing HomeHalf liked component with closed state', () => {
  const wrapper = shallow(<HomeHalf section="liked" chosen={false} />);
  expect(wrapper.find('.closed').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});
