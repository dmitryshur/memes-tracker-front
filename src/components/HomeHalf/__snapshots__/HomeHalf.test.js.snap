// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HomeHalf liked component 1`] = `
<div
  className="home-half home-half-liked closed"
>
  <div
    className="home-overlay home-overlay-liked"
  />
  <Connect(HalfContent)
    section="liked"
  />
</div>
`;

exports[`testing HomeHalf liked component with closed state 1`] = `
<div
  className="home-half home-half-liked closed"
>
  <div
    className="home-overlay home-overlay-liked"
  />
  <Connect(HalfContent)
    section="liked"
  />
</div>
`;

exports[`testing HomeHalf liked component with default state 1`] = `
<div
  className="home-half home-half-liked closed"
>
  <div
    className="home-overlay home-overlay-liked"
  />
  <Connect(HalfContent)
    section="liked"
  />
</div>
`;

exports[`testing HomeHalf liked component with opened state 1`] = `
<div
  className="home-half home-half-liked opened"
>
  <div
    className="home-overlay home-overlay-liked"
  />
  <Connect(HalfContent)
    section="liked"
  />
</div>
`;

exports[`testing HomeHalf newest component 1`] = `
<div
  className="home-half home-half-newest closed"
>
  <div
    className="home-overlay home-overlay-newest"
  />
  <HalfTitles />
  <Connect(HalfContent)
    section="newest"
  />
</div>
`;

exports[`testing HomeHalf newest component with closed state 1`] = `
<div
  className="home-half home-half-newest closed"
>
  <div
    className="home-overlay home-overlay-newest"
  />
  <HalfTitles />
  <Connect(HalfContent)
    section="newest"
  />
</div>
`;

exports[`testing HomeHalf newest component with default state 1`] = `
<div
  className="home-half home-half-newest closed"
>
  <div
    className="home-overlay home-overlay-newest"
  />
  <HalfTitles />
  <Connect(HalfContent)
    section="newest"
  />
</div>
`;

exports[`testing HomeHalf newest component with opened state 1`] = `
<div
  className="home-half home-half-newest opened"
>
  <div
    className="home-overlay home-overlay-newest"
  />
  <HalfTitles />
  <Connect(HalfContent)
    section="newest"
  />
</div>
`;
